src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java

```diff
@@ -332,10 +343,22 @@ public static FeedSource checkFeedSourcePermissions(Request req, FeedSource feed
             // Throw halt if user not authorized.
             logMessageAndHalt(req, 403, "User not authorized to perform action on feed source");
         }
+
+        // Remove labels user is not allowed to see if user is not admin
+        if (!isAdmin) {
+            feedSource.labels = feedSource.labels.stream()
+                    // Need to resolve label IDs to labels, then back
+                    .map(labelId -> Persistence.labels.getById(labelId))
```

Comment:
It's a little inefficient to be making multiple individual queries for a single label like this.

Reply:
Disregard if refactoring as suggested in https://github.com/ibi-group/datatools-server/pull/403#pullrequestreview-714190079.